prusti-interface/src/environment/mir_dump/lifetimes.rs

```diff
@@ -45,6 +49,31 @@ impl Lifetimes {
             output_facts,
         }
     }
+    pub fn get_loan_live_at_start(&self, location: mir::Location) -> BTreeSet<String> {
+        let info = self.get_loan_live_at(RichLocation::Start(location));
+        info.into_iter()
+            .map(|x| opaque_lifetime_string(x.index()))
+            .collect()
+    }
+    pub fn get_origin_contains_loan_at_mid(
+        &self,
+        location: mir::Location,
+    ) -> BTreeMap<String, BTreeSet<String>> {
+        let info = self.get_origin_contains_loan_at(RichLocation::Mid(location));
+        info.iter()
+            .map(|(k, v)| {
+                (
+                    k.lifetime_string(),
+                    v.iter()
+                        .map(|x| opaque_lifetime_string(x.index()))
+                        .collect(),
+                )
+            })
+            .collect()
+    }
+    pub fn edge_count(&self) -> u32 {
```

Comment:
Why `u32` and not `usize`? In general, it is better to use `try_into()` instead of `as` for casting that may fail.

Reply:
`u32` doesn't seem to work in the Reference in `rvalue.rs`. It can't lower it to_middle, not quite sure why `u32` works and `usize` doesn't.

Reply:
I see. Then at least use `try_into().unwrap()` to detect in case something goes wrong.

Reply:
ok, done